src/components/compress/index.tsx

```diff
@@ -100,12 +100,22 @@ async function preprocessImage(
 ): Promise<ImageData> {
   let result = source.processed;
 
+  if (isWorkerResizeOptions(preprocessData.resize) && isHqx(preprocessData.resize)) {
+    result = await processor.workerResize(result, preprocessData.resize);
+  }
   if (preprocessData.resize.enabled) {
     if (preprocessData.resize.method === 'vector' && source.vectorImage) {
       result = processor.vectorResize(
         source.vectorImage,
         preprocessData.resize,
       );
+    } else if (isHqx(preprocessData.resize)) {
+      // Hqx can only do x2, x3 or x4. That part has already been done above.
+      // If the target size is not a clean x2, x3 or x4, use nearest neighbor
+      // for the remaining scaling.
+      const pixelOpts = { ...preprocessData.resize, method: 'browser-pixelated' };
```

Comment:
Let's switch to one of the other resize methods. If we're going to be downscaling, it seems like catmull-rom does well.

Also, we can skip resizing if it's already the target size.

Reply:
Hah, just saw you came to the same conclusion in terms of catmull-rom.